File: FinanceAPI/FinanceAPI/DTOs/ExpenseDto.cs
using System.ComponentModel.DataAnnotations;

namespace FinanceAPI.DTOs
{

    public class CreateExpenseDto
    {
        public int WalletId { get; set; }
        public string? Description { get; set; }
        [Required]
        public int Amount { get; set; }
        [Required]
        public int CategoryId { get; set; }
    }

    public class ExpenseDto
    {
        public int Id { get; set; }
        public string Description { get; set; }
        public int Amount { get; set; }
        public DateTime Date { get; set; }
        public string CategoryName { get; set; }
        public int CategoryId { get; set; }
        public int WalletId { get; set; }
    }

    public class UpdateExpenseDto
    {
        public string? Description { get; set; }
        [Required]
        public int Amount { get; set; }
        [Required]
        public int CategoryId { get; set; }
    }
}

File: FinancerClient/src/app/model/ExpenseDto.ts
export interface ExpenseDto {
  id: number;
  description: string;
  amount: number;
  date: string; 
  categoryName: string;
  categoryId: number;
  walletId: number;
}

export interface CreateExpenseDto {
  walletId: number;
  description?: string;
  amount: number;
  categoryId: number;
}

export interface UpdateExpenseDto {
  description?: string;
  amount: number;
  categoryId: number;
}
